sections/shared/Layout/AppLayout/Header/UserMenu/UserMenu.tsx

```diff
@@ -56,6 +59,25 @@ const UserMenu: FC<UserMenuProps> = ({ isTextButton }) => {
 		[redeemableDeprecatedSynths?.totalUSDBalance]
 	);
 
+	/**
+	 * @dev We require this new provider since we need one connected to Ethereum
+	 * mainnet and NOT to the Optimism network.
+	 */
+	const infuraUrl = 'https://mainnet.infura.io/v3/';
+
+	let ensProvider = new ethers.providers.JsonRpcProvider(
+		infuraUrl + process.env.NEXT_PUBLIC_INFURA_PROJECT_ID
+	);
+
+	useEffect(() => {
+		if (signer) {
+			let account: any = signer.getAddress();
+			getENSNameAndAvatarUrl(account, ensProvider).then((ensObj) => {
+				if (ensObj !== null) setEns(ensObj?.name);
+			});
+		}
+	}, [signer]);
```

Comment:
if you still don't want to add a useEffect hook inside of `UserMenu.tsx`, i understand

just lmk 😃

Reply:
I think this is okay for now. I don't think we show a user's address anywhere else.